src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -199,7 +199,17 @@ def close(self) -> None:
         self.__closed = True
 
     def join(self, timeout: Optional[float] = None) -> None:
-        self.__flush()
+        start = time.time()
```

Comment:
Curious to know what timeout value is passed over here.

Another thing to consider is that join's are chained between multiple strategies. I am not sure if the timeout value decreases during chained joins or the timeout is fixed for all join operations.

Reply:
@nikhars that's a good point. I actually realize no timeout is passed along at all, not sure if that's by design from the processor and that each step determines it's own timeout (and what gets passed along). @fpacifici @lynnagara do you have context here?

Reply:
> I actually realize no timeout is passed along at all, not sure if that's by design from the processor 

I thought we were passing it. if that is not a good idea as it can get stuck forever.